wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -36,11 +36,20 @@ import kotlin.reflect.KClass
 
 expect abstract class ProtoAdapter<E>(
   fieldEncoding: FieldEncoding,
-  type: KClass<*>?
+  type: KClass<*>?,
+  typeUrl: String?
```

Comment:
This will break binary compat, need to introduce a new constructor that passes a null.

Reply:
This will break for JS/Native but we keep it for JVM which is the only one we're caring about for the moment.

Reply:
Not sure I understand what you mean by "we keep it for JVM". This change will break JVM too.

Reply:
We added them on the JVM https://github.com/square/wire/pull/1391/files#diff-5a46a66f7227b9d77ce0769f7393d28cR46-R51

Reply:
Ah, I see. Why not add them in common to not break other platforms?

Reply:
Didn't feel it was worth it. Nobody really uses the other platform except protogram I think.

Reply:
and there’s no binary compatibility there anyway